### ISSUE TITLE
Firefox Relay does not work at all (with Yahoo Mail?)

### ISSUE THREAD

**Issue by cadeyrn** [mozilla/fx-private-relay#478]
Unfortunately Firefox Relay does not work at all for me. I tried it several times in the last few days. I created new relay addresses via the dashboard, I sent e-mails from another mail account to the relay addresses, I also tried it via a registration form in a forum, but no e-mail was ever forwarded. Also the dashboard still shows 0 forwarded e-mails.

Since Firefox Relay is still _invitation only_ I can't check whether it works with another e-mail provider or not. My relay addresses should forward messages to a Yahooo Mail account (@ymail.com). Maybe you could check with a @ymail.com address if you can reproduce the problem. Let me know if / how I can help to debug the issue.

**Comment by gaby2300** [comment]
I have the same issue with Gmail.

**Comment by groovecoder** [comment]
I'll be looking into this today. Do you mind saying specifically - from WHERE did you try to send emails to the relay addresses? The sender MTA makes a big difference.

**Comment by cadeyrn** [comment]
> Do you mind saying specifically - from WHERE did you try to send emails to the relay addresses? The sender MTA makes a big difference.

From Google Mail (@googlemail.com) to Yahoo Mail (@ymail.com).

**Comment by groovecoder** [comment]
Thanks. I'll try a bunch of different combinations of sending + receiving MTAs.

**Comment by deimidis** [comment]
I had it working until Saturday, now these days I didn't receive mails, and didn't it blocked on Relay Dashboard. I use it for a newsletter that came from mailchimp, I think. My email is a protonmail email with a personal domain. Let me know if it can be the same problem and what other data you need. Thanks!

**Comment by groovecoder** [comment]
Matrix of tests to run/try:

* [x] Gmail -> Relay -> Gmail ✅ 
* [x] Yahoo -> Relay -> Gmail ✅ 
* [x] Hotmail -> Relay -> Gmail ✅ 
* [x] Proton -> Relay -> Gmail ✅ 
* [ ] Gmail -> Relay -> Yahoo
* [ ] Yahoo -> Relay -> Yahoo
* [ ] Hotmail -> Relay -> Yahoo
* [ ] Proton -> Relay -> Yahoo
* [ ] Gmail -> Relay -> Hotmail
* [ ] Yahoo -> Relay -> Hotmail
* [ ] Hotmail -> Relay -> Hotmail
* [ ] Proton -> Relay -> Hotmail
* [ ] Gmail -> Relay -> Proton
* [ ] Yahoo -> Relay -> Proton
* [ ] Hotmail -> Relay -> Proton
* [ ] Proton -> Relay -> Proton

If anyone can test and add notes and comments here that'd be great.

**Comment by pdehaan** [comment]
Not very helpful, but a random Google search to a random article gave this list:

> ### THE BEST EMAIL SERVICE
> 1. ProtonMail
> 2. Gmail
> 3. Outlook
> 4. Yahoo Mail
> 5. Zoho

Looks like you already have Proton, Gmail, Yahoo! in your matrix. Not sure if it's worth adding Zoho and Outlook (although maybe thats the same as Hotmail, I can't remember if those are the same site now or still subtly different). Still trying to find some definitive list of most popular email providers in 2020 (that aren't your ISP or whatever).

**Comment by groovecoder** [comment]
AFAICT Outlook and Hotmail are the same thing now, just with different addresses.

**Comment by groovecoder** [comment]
@cadeyrn @gaby2300 @deimidis - can you check your spam folders too?

**Comment by gaby2300** [comment]
@groovecoder nothing in Spam either

**Comment by groovecoder** [comment]
@pdehaan and I have spot-checked everything except proton -> relay -> proton. Can you test again? We may have had a transient/intermittent deliver-ability problem.

**Comment by cadeyrn** [comment]
Tested again. Nothing in the spam folder, still "0 forwared" on the dashboard.

**Comment by groovecoder** [comment]
Okay, it sounds like this "invalid signature" error we're seeing every once in a while coincided with your test email, which makes me 😢 because I hate debugging crypto code.

This was from a @googlemail.com address to a @ymail.com address?

**Comment by groovecoder** [comment]
I'm not sure how to get a @ymail.com address ? My yahoo email address ends with @yahoo.com.

**Comment by cadeyrn** [comment]
> This was from a @googlemail.com address to a @ymail.com address?

Yes!

> I'm not sure how to get a @ymail.com address ? My yahoo email address ends with @yahoo.com.

I don't know if it's still possible to register @ymail.com addresses. It's a very old e-mail address. 😅

**Comment by deimidis** [comment]
> @cadeyrn @gaby2300 @deimidis - can you check your spam folders too?

I searched for it (beasue protonmail spam filters are strange) but nothing. Other thing that happened, configurations emails with mailchimp and the newsletter worked, but still didn't received the newsletter. I sent an email to @gaby2300 relay address and worked.

Maybe it's not the email itself, just something mailchimp is doing when it send the newsletter?

**Comment by groovecoder** [comment]
Yeah, different senders send with different content types and parsing multi-part emails is particularly tricky.

Do you know if the emails that are failing to send have attachments?

**Comment by cadeyrn** [comment]
In my case the emails didn't have attachments.

**Comment by groovecoder** [comment]
Thanks for the input. We've fixed a couple of deliverability bugs on the dev server and we should be able to push those to production on Monday, so we may ask you to re-test again.

**Comment by groovecoder** [comment]
We just deployed updates that should improve handling emails with `multipart` and with email senders with different character encodings. Can you check some of the email deliveries again?

**Comment by cadeyrn** [comment]
It worked for Gmail -> Ymail. \o/

But Firefox Relay messed up the Sender's Name

From: "\"Sï¿½ren Hentzschel\" <…@googlemail.com>

Should I file a new issue?

**Comment by groovecoder** [comment]
Glad it worked! Yes please file an additional issue for that. And can you include what the expected Sender's name should be? (The bug had to do with python 2 vs. 3 character encodings, so I imagine we'll have to dig in deeper to fix it.)

**Comment by cadeyrn** [comment]
I filed #523 for the other issue.
